### PR TITLE
feat: export `DialogSurfaceMotion`

### PR DESCRIPTION
**Pull request by wbarrantes**
## Previous Behavior

`DialogSurfaceMotion` is not exported from `@fluentui/react-components`.

## New Behavior

`DialogSurfaceMotion` is exported from `@fluentui/react-components`.

## Context

Having access to the `DialogSurfaceMotion` would be beneficial for implementing an `onAfterClose` event or similar events. For instance: https://stackblitz.com/edit/f6n23z?file=src%2Fexample.tsx

### CODE REVIEW

**Review by spmonahan — APPROVED** [review]

**Review by bsunderhus — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">38</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">29</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Avatar-mount_base.html">650</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Avatar-mount.html">645</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Button-mount_base.html">306</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Button-mount.html">332</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Field-mount_base.html">1117</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Field-mount.html">1191</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProvider-mount_base.html">721</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProvider-mount.html">729</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-mount_base.html">86</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">38</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">29</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/MakeStyles-mount_base.html">885</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/MakeStyles-mount.html">860</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Persona-mount_base.html">1724</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/Persona-mount.html">1719</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/SpinButton-mount_base.html">1440</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/SpinButton-mount.html">1404</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/SwatchPicker-mount_base.html">1693</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33065/perf-test-react-components/SwatchPicker-mount.html">1704</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-components-2d388f43-423c-43ec-9a07-c05f29d0651c.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-components-2d388f43-423c-43ec-9a07-c05f29d0651c.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33065&commitId=f1a3a9e9d4e7534e5e3e5f72451f41761237fb31&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.chromium.png|14| Changed|
|Avatar Converged.Badge Mask RTL.chromium.png|4| Changed|

</details>


  <details><summary><b>Drawer</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay High Contrast.chromium.png|2251| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by layershifter — CHANGES_REQUESTED** [review]
@wbarrantes it's an internal implementation and we would like to avoid it being exported. There are two ways to solve your problem:
- Just pass a callback i.e. `<Dialog surfaceMotion={{ onMotionFinish }} />`
- Use a render function i.e.:
   ```
    <Dialog
      surfaceMotion={{
        children: (Component, props) => (
          <Component onMotionFinish={onMotionFinish} {...props}>
            {props.children}
          </Component>
        ),
      }}
    />
    ```

**Comment by spmonahan** [comment]
> @wbarrantes it's an internal implementation and we would like to avoid it being exported. There are two ways to solve your problem:
> 
> * Just pass a callback i.e. `<Dialog surfaceMotion={{ onMotionFinish }} />`
> * Use a render function i.e.:
>   ```
>    <Dialog
>      surfaceMotion={{
>        children: (Component, props) => (
>          <Component onMotionFinish={onMotionFinish} {...props}>
>            {props.children}
>          </Component>
>        ),
>      }}
>    />
>   ```

In that case, assuming this works for @wbarrantes , let's change this PR to add this to the docs.

**Pull request closed without merging**